.github/workflows/discover-lean-pr-testing.yml

```
@@ -146,13 +146,44 @@ jobs:
         for BRANCH in $BRANCHES; do
           # Extract PR number from branch name
           PR_NUMBER=$(echo "$BRANCH" | grep -oP '\d+$')
-          GITHUB_DIFF="https://github.com/leanprover-community/mathlib4/compare/nightly-testing...lean-pr-testing-$PR_NUMBER"
+
+          # Make sure we're on nightly-testing branch before doing fetch operations
+          git checkout nightly-testing
+
+          # Fetch all tags in the repository
+          git fetch --tags
+
+          # Fetch the PR branch
+          git fetch origin "$BRANCH"
+
+          # Find the most recent nightly-testing-YYYY-MM-DD tag that is an ancestor of the branch
+          # Temporarily checkout the branch
+          git checkout --quiet origin/"$BRANCH" || {
```

```suggestion
          git checkout origin/"$BRANCH" || {
```
why go `quiet` if you also produce a custom error message on failure?  Besides, the message that git gives may be useful for debugging once something unexpected happens